src/nvim/event/process.c

```diff
@@ -120,14 +120,10 @@ void process_teardown(Loop *loop) FUNC_ATTR_NONNULL_ALL
       // Close handles to process without killing it.
       CREATE_EVENT(loop->events, process_close_handles, 1, proc);
     } else {
-      if (proc->type == kProcessTypeUv) {
-        uv_kill(proc->pid, SIGTERM);
-        proc->term_sent = true;
-        process_stop(proc);
-      } else {  // kProcessTypePty
-        process_close_streams(proc);
-        pty_process_close_master((PtyProcess *)proc);
-      }
+      // Make sure the job kill timer is running and all children are marked
+      // for termination. Otherwise children could block us forever in the
+      // following event loop.
+      process_stop(proc);
```

Comment:
The old behavior was intentional, a posix pty is not supposed to kill the terminal's child processes, just closing the master should send SIGHUP automatically and let the child act accordingly. If we're waiting somewhere for the pty job to die that wait should just be removed.